internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_machine_instance_resource.go

```diff
@@ -0,0 +1,986 @@
+}
+
+type Hardware struct {
+	CpuCount                    int64 `tfschema:"cpu_count"`
+	DynamicMemoryMaxInMb        int64 `tfschema:"dynamic_memory_max_in_mb"`
+	DynamicMemoryMinInMb        int64 `tfschema:"dynamic_memory_min_in_mb"`
+	LimitCpuForMigrationEnabled bool  `tfschema:"limit_cpu_for_migration_enabled"`
+	MemoryInMb                  int64 `tfschema:"memory_in_mb"`
+}
+
+type Infrastructure struct {
+	CheckpointType                                          string `tfschema:"checkpoint_type"`
+	SystemCenterVirtualMachineManagerCloudId                string `tfschema:"system_center_virtual_machine_manager_cloud_id"`
+	SystemCenterVirtualMachineManagerInventoryItemId        string `tfschema:"system_center_virtual_machine_manager_inventory_item_id"`
+	SystemCenterVirtualMachineManagerTemplateId             string `tfschema:"system_center_virtual_machine_manager_template_id"`
+	SystemCenterVirtualMachineManagerVirtualMachineServerId string `tfschema:"system_center_virtual_machine_manager_virtual_machine_server_id"`
+}
+
+type NetworkInterface struct {
+	Name             string `tfschema:"name"`
+	Ipv4AddressType  string `tfschema:"ipv4_address_type"`
+	Ipv6AddressType  string `tfschema:"ipv6_address_type"`
+	MacAddressType   string `tfschema:"mac_address_type"`
+	VirtualNetworkId string `tfschema:"virtual_network_id"`
+}
+
+type OperatingSystem struct {
+	ComputerName  string `tfschema:"computer_name"`
+	AdminPassword string `tfschema:"admin_password"`
+}
+
+type StorageDisk struct {
+	Bus                  int64  `tfschema:"bus"`
+	BusType              string `tfschema:"bus_type"`
+	DiskSizeGB           int64  `tfschema:"disk_size_gb"`
+	Lun                  int64  `tfschema:"lun"`
+	Name                 string `tfschema:"name"`
+	StorageQoSPolicyName string `tfschema:"storage_qos_policy_name"`
+	TemplateDiskId       string `tfschema:"template_disk_id"`
+	VhdType              string `tfschema:"vhd_type"`
+}
+
+var (
+	_ sdk.Resource           = SystemCenterVirtualMachineManagerVirtualMachineInstanceResource{}
+	_ sdk.ResourceWithUpdate = SystemCenterVirtualMachineManagerVirtualMachineInstanceResource{}
+)
+
+type SystemCenterVirtualMachineManagerVirtualMachineInstanceResource struct{}
+
+func (r SystemCenterVirtualMachineManagerVirtualMachineInstanceResource) ModelObject() interface{} {
+	return &SystemCenterVirtualMachineManagerVirtualMachineInstanceModel{}
+}
+
+func (r SystemCenterVirtualMachineManagerVirtualMachineInstanceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.SystemCenterVirtualMachineManagerVirtualMachineInstanceID
+}
+
+func (r SystemCenterVirtualMachineManagerVirtualMachineInstanceResource) ResourceType() string {
+	return "azurerm_system_center_virtual_machine_manager_virtual_machine_instance"
+}
+
+func (r SystemCenterVirtualMachineManagerVirtualMachineInstanceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"scoped_resource_id": {
```

Comment:
it looks like this is always an `azurerm_arc_machine` id - should we consider changing this to to `arc_machine_id` for clarity?

Reply:
To anticipate the service team supporting more types of resourceUri in the future, I defined it with a generic name. Additionally, I noticed that existing resources in Terraform are defined this way, so I followed their approach.